test/rom_test.c

```diff
@@ -0,0 +1,53 @@
+#include <criterion/criterion.h>
+#include <criterion/new/assert.h>
+#include <criterion/logging.h>
+#include <criterion/redirect.h>
+#include <string.h>
+#include <libgen.h>
+
+#include "rom.h"
+#include "log.h"
+
+#define MAX_PATH_LENGTH    (512)
+#define MAX_LOG_MSG_LENGTH (512)
+
+Test(rom, rom_init_file_not_found, .exit_code = EXIT_FAILURE, .init = cr_redirect_stderr) {
+    rom_init("some/weird/path/yobemag.gb");
+}
+
+Test(rom, rom_init_file_found, .exit_code = EXIT_SUCCESS, .init = cr_redirect_stderr) {
+    char *file_path_copy                = strdup(__FILE__);
+    char rom_file_path[MAX_PATH_LENGTH] = {0};
+    snprintf(rom_file_path, MAX_PATH_LENGTH, "%s/../roms/yobemag.gb", dirname(file_path_copy));
+
+    cr_assert(get_rom_bytes() == NULL);
```

Comment:
```suggestion
    cr_expect_null(get_rom_bytes());
```
may be a more eloquent function to use